models/optional_modules/consent/consent.yml

```diff
@@ -0,0 +1,134 @@
+      - name: domain_userid
+        description: '{{ doc("col_domain_userid") }}'
+      - name: user_id
+        description: The optional userid of a user
+      - name: geo_country
+        description: '{{ doc("col_geo_country") }}'
+      - name: page_view_id
+        description: '{{ doc("col_page_view_id") }}'
+      - name: domain_sessionid
+        description: '{{ doc("col_domain_sessionid") }}'
+      - name: derived_tstamp
+        description: '{{ doc("col_derived_tstamp") }}'
+      - name: event_name
+        description: '{{ doc("col_event_name") }}'
+      - name: event_type
+        description: The type of consent event. E.g allow_all
+      - name: basis_for_processing
+        description: A legal term that defines the basis for collecting consent to process data
+      - name: consent_url
+        description: The privacy policy URL for which the visitor can refer to
+      - name: consent_version
+        description: The version of the privacy policy.
+      - name: consent_scopes
+        description: The scopes that the visitor has consented to. The minimum length can be 1 if we suppose that you always have a part of necessary cookies or storage that needs to take be used for the website to work properly
+      - name: domains_applied
+        description: The domains for which the consent is valid
+      - name: elapsed_time
+        description: The time taken for the context box to be shown to the screen
+  - name: snowplow_web_consent_users
+    description: By user consent stats
+    columns:
+      - name: domain_userid
+        tests:
+            - unique
+            - not_null
+        description: '{{ doc("col_domain_userid") }}'
+      - name: user_id
+        description: The optional userid of a visitor
+      - name: geo_country
+        description: '{{ doc("col_geo_country") }}'
+      - name: cmp_events
+        description: The number of cmp_visible events the user has generated
+      - name: last_cmp_event_tstamp
+        description: The timestamp of the last cmp_visible event
+      - name: last_consent_event_tstamp
+        description: The timestamp of the last consent event after the cmp_visible event happened
+      - name: last_consent_event_type
+        description: The type of the last consent event after the cmp_visible event happened
+      - name: last_consent_scopes
+        description: The list of consent scopes in connection with the last consent event
+      - name: last_consent_version
+        description: The privacy policy version in connection with the last consent event
+      - name: last_consent_url
+        description: The privacy policy url in connection with the last consent event
+      - name: last_domains_applied
+        description: The domains for which the last consent event applies
+      - name: is_official_version
+        description: A boolean to filter whether the last consent or cmp visible event is sent after the latest privacy policy version goes live
```

Comment:
❓ Is this technically true? It's based off the users table and users first filters out cmp_visible events?

Reply:
Not, unless we instruct them to send a test event, where we do not care about losing the related cmp_visible event. I know this is very ugly. Still thinking with tracker team but if educated they should be good to go.